backend/src/main/java/com/google/minicrm/data/Campaign.java

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.minicrm.data;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Represents a Campaign and all of its data. Supports conversion to datastore Entity objects and
+ * back. Campaigns are direct children entities of the Advertiser entity that they belong to. They
+ * are not directly linked to their leads. Instead, leads are linked to their respective campaign
+ * with the campaign id instance variable.
+ */
+public final class Campaign implements DatastoreObject {
```

Comment:
Since most of this class is the same as the Form class, we probably want to create a superclass.

Reply:
Hm, I did consider this when I was writing it but wasn't sure if we wanted to do it at this time since we may change how the Form object and the Campaign object work in the future.
Also, I think if we want to create a superclass, it should be done in a future PR since there's quite a bit of changes to be made with structure and well as setting up custom GSON deserialization for it.

Reply:
Gotchu. I will post an issue for this then.